jetty-server/src/main/java/org/eclipse/jetty/server/session/FileSessionDataStore.java

```diff
@@ -242,24 +265,12 @@ public void sweepDisk()
     public void sweepFile(long now, Path p)
         throws Exception
     {
-        if (p == null)
-            return;
-
-        try
+        if (isExpired(now, p))
         {
-            long expiry = getExpiryFromFilename(p.getFileName().toString());
-            //files with 0 expiry never expire
-            if (expiry > 0 && ((now - expiry) >= (5 * TimeUnit.SECONDS.toMillis(_gracePeriodSec))))
-            {
-                Files.deleteIfExists(p);
-                if (LOG.isDebugEnabled())
-                    LOG.debug("Sweep deleted {}", p.getFileName());
-            }
-        }
-        catch (NumberFormatException e)
-        {
-            LOG.warn("Not valid session filename {}", p.getFileName());
-            LOG.warn(e);
+            if (!Files.deleteIfExists(p))
+                LOG.warn("Could not delete {}", p.getFileName());
+            if (LOG.isDebugEnabled())
```

Comment:
So the debug output looking like the delete happened will be done even if the file wasn't actually deleted.

Reply:
factored out to a better common deleteFile method